gap/generic/SnAnUnknownDegree.gi

```diff
@@ -0,0 +1,895 @@
+#   - n is the degree of the group.
+# - slpToStdGens: an SLP to stdGens.
+#
+# TODO: Image Computation requires n >= 11.
+RECOG.ConstructSnAnIsomorphism := function(ri, n, stdGensAnWithMemory)
+    local stdGensAn, xis, gImage, foundOddPermutation, slp, eval,
+        hWithMemory, bWithMemory, stdGensSnWithMemory, b, h, g;
+    stdGensAn := StripMemory(stdGensAnWithMemory);
+    xis := RECOG.ConstructXiAn(n, stdGensAn[1], stdGensAn[2]);
+    foundOddPermutation := false;
+    for g in ri!.gensHmem do
+        gImage := RECOG.FindImageAn(ri, n, StripMemory(g),
+                                    stdGensAn[1], stdGensAn[2],
+                                    xis[1], xis[2]);
+        if gImage = fail then return fail; fi;
+        if SignPerm(gImage) = -1 then
+            # we found an odd permutation,
+            # so the group cannot be An
+            foundOddPermutation := true;
+            break;
+        fi;
+        slp := RECOG.SLPforAn(n, gImage);
+        eval := ResultOfStraightLineProgram(slp,
+                                            [stdGensAn[2], stdGensAn[1]]);
+        if not isequal(ri)(eval, StripMemory(g)) then return fail; fi;
+    od;
+    if not foundOddPermutation then
+        return rec(type := "An",
+                   isoData := [[stdGensAn[1], stdGensAn[2]], xis, n],
+                   slpToStdGens := SLPOfElms(stdGensAnWithMemory));
+    fi;
+    # Construct standard generators for Sn: [bWithMemory, hWithMemory].
+    slp := RECOG.SLPforAn(n, (1,2) * gImage);
+    eval := ResultOfStraightLineProgram(
+        slp, [stdGensAnWithMemory[2], stdGensAnWithMemory[1]]
+    );
+    hWithMemory := eval * g ^ -1;
+    bWithMemory := stdGensAnWithMemory[1] * stdGensAnWithMemory[2];
+    if n mod 2 = 0 then
+        bWithMemory := hWithMemory * bWithMemory;
+    fi;
+    stdGensSnWithMemory := [bWithMemory, hWithMemory];
+    b := StripMemory(bWithMemory);
+    h := StripMemory(hWithMemory);
+    if not RECOG.SatisfiesSnPresentation(ri, n, b, h) then
+        return fail;
+    fi;
+    xis := RECOG.ConstructXiSn(n, b, h);
+    for g in ri!.gensHmem do
+        gImage := RECOG.FindImageSn(ri, n, StripMemory(g),
+                                    b, h,
+                                    xis[1], xis[2]);
+        if gImage = fail then return fail; fi;
+        slp := RECOG.SLPforSn(n, gImage);
+        eval := ResultOfStraightLineProgram(slp, [h, b]);
+        if not isequal(ri)(eval, StripMemory(g)) then return fail; fi;
+    od;
+    return rec(type := "Sn",
+               isoData := [[b, h], xis, n],
+               slpToStdGens := SLPOfElms(stdGensSnWithMemory));
+end;
+
+# This method is an implementation of <Cite Key="JLNP13"/>. It is the main
+# function of SnAnUnknownDegree.
+# Note that it currently only works for 11 <= <A>n</A>. To make it work with
+# smaller <A>n</A>, we need to include fixes from Jonathan Conder's B.Sc.
+# Thesis "Algorithms for Permutation Groups".
```

Comment:
So... are these fixes included? If so: where are they? I searched for comments referencing "Conder" but saw none... Or this more a "TODO" comment?

Reply:
Aha, I just saw that these are implemented in PR #265

Reply:
Yes, this should be changed to a "TODO" comment.